roles/ohpc_matlab/tasks/main.yaml

```diff
@@ -0,0 +1,24 @@
+- name: install the latest version of libXtst
+  yum:
+    name: libXt
+    state: latest
+
+- name: Download matlab
+  get_url:
+    url: "{{ matlab_download_url }}"
+    dest: "{{ matlab_destination  }}"
+
+- name: Extract matlab
+  unarchive:
+    src: "{{ matlab_destination  }}"
+    dest: "{{ matlab_clustershare }}"
+
+- file: Create directory
+    path: /export/eb/modules/all/matlab
+    state: directory
+    mode: 0755
+
+- name: Copy modulefile from template to /export/eb/modules/all/matlab/ 
+  template:
```

Comment:
The src should be a variable and the destination should be composed of two variables.  Also the name should reference the variable location an not an explicit string.